schema-rules/src/test/java/io/confluent/kafka/schemaregistry/rules/cel/CelExecutorTest.java

```diff
@@ -393,6 +406,41 @@ public void testKafkaAvroSerializerReflectionFieldTransformNoTags() throws Excep
     assertEquals("678-suffix", ((OldWidget)obj).getPiiMap().get("key2").getPii());
   }
 
+  @Test
+  public void testKafkaAvroSerializerReflectionFieldTransformWithGuard() throws Exception {
+    byte[] bytes;
+    Object obj;
+
+    OldWidget widget = new OldWidget("alice");
+    widget.setSsn(ImmutableList.of("123", "456"));
+    widget.setPiiArray(ImmutableList.of(new OldPii("789"), new OldPii("012")));
+    widget.setPiiMap(ImmutableMap.of("key1", new OldPii("345"), "key2", new OldPii("678")));
+    Schema schema = createWidgetSchemaWithGuard();
+    AvroSchema avroSchema = new AvroSchema(schema);
+    Rule rule = new Rule("myRule", null, RuleKind.TRANSFORM, RuleMode.WRITE,
+        CelFieldExecutor.TYPE, null, null, "typeName == 'STRING'; value + \"-suffix\"",
+        null, null, false);
+    RuleSet ruleSet = new RuleSet(Collections.emptyList(), Collections.singletonList(rule));
+    avroSchema = avroSchema.copy(null, ruleSet);
+    schemaRegistry.register(topic + "-value", avroSchema);
+
+    bytes = reflectionAvroSerializer.serialize(topic, widget);
+
+    obj = reflectionAvroDeserializer.deserialize(topic, bytes, schema);
+    assertTrue(
+        "Returned object should be a Widget",
+        OldWidget.class.isInstance(obj)
+    );
+    assertEquals(widget, obj);
+    assertEquals("alice-suffix", ((OldWidget)obj).getName());
+    assertEquals("123-suffix", ((OldWidget)obj).getSsn().get(0));
+    assertEquals("456-suffix", ((OldWidget)obj).getSsn().get(1));
+    assertEquals("789-suffix", ((OldWidget)obj).getPiiArray().get(0).getPii());
+    assertEquals("012-suffix", ((OldWidget)obj).getPiiArray().get(1).getPii());
+    assertEquals("345-suffix", ((OldWidget)obj).getPiiMap().get("key1").getPii());
+    assertEquals("678-suffix", ((OldWidget)obj).getPiiMap().get("key2").getPii());
```

Comment:
also check `size` and `version` don't have the `-suffix`?

Reply:
Thanks, I'll add